### PR TITLE
[FIXED JENKINS-28111] JNLP slave JVM Options are inconsistently applied

### PR DESCRIPTION
**Pull request by stephenc**
@reviewbybees

### CODE REVIEW

**Comment by jtnord** [comment]
shouldn't the args be inside `java-vm-args` in the [JNLP](https://docs.oracle.com/javase/tutorial/deployment/deploymentInDepth/jnlpFileSyntax.html) file?

**Comment by stephenc** [comment]
@jtnord and they are. This is to stop confusion if you copy and paste the "java -jar ..." command

**Comment by stephenc** [comment]
@jtnord if you use the "java -jar slave.jar ..." version of the command then it just uses the jnlp url to pull out some stuff but it doesn't spawn a new JVM, so the JVM options are those that you launch `java` with... hence we should put the JVM options onto that command, i.e. this issue

**Comment by stephenc** [comment]
@jtnord if you use the `javaws` version of the command, everything is peachy. Same goes with the `Launch` button.

**Comment by jtnord** [comment]
So we are doing a poor mans re-implementation of javaws because the end user may not have it installed as it's not in the server JRE   :-/

:+1:

**Comment by stephenc** [comment]
@jtnord it's not a reimplementation of javaws... it's worse than that: it just uses the URL and resolves Jenkins from that URL... but I'm not going to go changing all that when a simple printing out of the JVM options is good enough to resolve this!

**Comment by stephenc** [comment]
CI Test failures are unrelated

File: core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly [hunk 1]
```diff
@@ -57,7 +57,7 @@ THE SOFTWARE.
                 <p>
                   ${%Or if the slave is headless:}
                 </p>
-                <pre>java -jar <a href="${rootURL}/jnlpJars/slave.jar">slave.jar</a> -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp</pre>
+                <pre>java ${it.launcher.vmargs}${it.launcher.vmargs==null?'':' '}-jar <a href="${rootURL}/jnlpJars/slave.jar">slave.jar</a> -jnlpUrl ${h.inferHudsonURL(request)}${it.url}slave-agent.jnlp</pre>
```
Comment (jglick) [inline core/src/main/resources/hudson/slaves/JNLPLauncher/main.jelly]: Less confusing to write

    ${it.launcher.vmargs == null ? '' : ' ' + it.launcher.vmargs}

And did you not want a trailing space before `-jar`?

**Comment by jglick** [comment]
:+1: 

BTW as a courtesy to reviewers please include a hyperlink to JIRA from the PR. Also make sure the JIRA issue is assigned to you, _In Progress_, and includes a link to the PR (I use the _Web Link_ option so it appears in main metadata rather than comments).

**Pull request merged**